modules/doc/content/application_development/profiling.md

```diff
@@ -1,6 +1,177 @@
+`[proc#].prof` to generate an independent profiling data file for each MPI
+rank/process. The above example would generate files `run1_0.prof`,
+`run1_1.prof`, `run1_2.prof`, ..., `run1_32.prof`.
+
+### Analyzing Profile Data
+
+Profiling data can be analyzed using the
+[pprof](https://github.com/google/pprof) utility which is included in the
+latest MOOSE environment package.  Or you can also build/install it yourself.
+When using `pprof`, the **exact same** compiled version of the binary you used
+to create the profile **must** still be located where it was when the profile
+was created. `pprof` presents two types of profiling times - "flat" and
+"cumulative".  "flat" times report the amount of time spent *directly* inside
+a function (i.e. excluding time spent in transitively called functions), and
+"cumulative" times report the complete amount of time spent in a function
+*including its descendants* (i.e. other functions called by the function
+transitively).
+
+pprof has an interactive mode accessed by passing a profiling file to the
+command: `pprof run1.prof`. In interactive mode, you can run several commands
+to interrogate the profiling data. A few of these commands are described
+below, but there are others and several options you can see by entering `help`
+or `help <cmd|option>` in interactive mode or by running `pprof --help` on the
+command line.  The following sections assume you are running in pprof's
+interactive mode.
+
+### top N
+
+`top [N]` shows the top N functions that used the most "flat" time:
+
+```
+(pprof) top 12
+Showing nodes accounting for 2030ms, 31.92% of 6360ms total
+Dropped 362 nodes (cum <= 31.80ms)
+Showing top 12 nodes out of 313
+      flat  flat%   sum%        cum   cum%
+     260ms  4.09%  4.09%      260ms  4.09%  (anonymous namespace)::fe_lagrange_1D_shape
+     230ms  3.62%  7.70%      250ms  3.93%  libMesh::DofObject::start_idx
+     210ms  3.30% 11.01%      240ms  3.77%  std::_Rb_tree::_M_lower_bound
+     200ms  3.14% 14.15%      430ms  6.76%  MooseVariableData::computeValues
+     200ms  3.14% 17.30%      200ms  3.14%  libMesh::Elem::active
+     170ms  2.67% 19.97%      170ms  2.67%  FEProblemBase::setNeighborSubdomainID
+     150ms  2.36% 22.33%      240ms  3.77%  _int_malloc
+     140ms  2.20% 24.53%      150ms  2.36%  libMesh::Elem::which_child_am_i
+     130ms  2.04% 26.57%      130ms  2.04%  std::vector::size
+     120ms  1.89% 28.46%      120ms  1.89%  std::vector::operator[] (inline)
+     110ms  1.73% 30.19%      200ms  3.14%  _int_free
+     110ms  1.73% 31.92%      220ms  3.46%  libMesh::Elem::level
+(pprof)
+```
+
+### svg and png
+
+`svg` and `png` create a visual profiling graph showing function call
+relationships and time spent in each function.  This is one of the most
+useful ways to look at the profiling data. The output is printed to screen
+by default - so you will want to redirect it to a file:
+
+```
+(pprof) png > run1.png
```

Comment:
It may not be clear out of context what this command is showing (e.g. the iteractive pprof prompt). Perhaps you should also mention how to run pprof with the png option directly and avoid showing people how to run it through the interactive mode?

Reply:
I'll add in the command that starts pprof interactive mode in front.  Interactive mode is super useful and is how I spend much of my time investigating profiles.  This isn't really meant to be a full guide for all the ways you can use pprof - gperftools and pprof have their own documentation for all the details.

Reply:
For some reason - github won't let me comment directly on your env var comment.  I really think the examples make it abundantly clear when and how to use them to turn profiling on.